

# **Plotly: Data Visualization in Python -**
* Line, Bar, Scatter, Histogram, Violin, Gantt, Heatmap, 3D Graphs.

## Introduction

* Plotly is a powerful Python visualization library for creating **interactive** and **publication-quality** graphs.
* It supports various plot types, making it ideal for **data exploration and storytelling**.
* For this case study, we analyze **Tourism & Travel Analytics**, focusing on trends in visitor numbers, seasonal patterns, hotel bookings, and travel costs.

## Plotly: Interactive Visualizations for Tourism & Travel Analytics

### Line Plot

#### Concept

* A **line plot** is used to visualize trends over time, making it ideal for **time-series data** like monthly tourist arrivals.

In [2]:
import plotly.express as px
import pandas as pd

df = pd.read_csv("tourism_data.csv")
fig = px.line(df, x='Month', y='Tourist_Arrivals', title="Monthly Tourist Arrivals")
fig.show()



# **Inference:**

* The graph helps **identify seasonal trends**.
* Peaks may indicate **holiday seasons**, while dips suggest **off-seasons**.
* Between May to September is peak season for tourist.
* From October to December it is declining and lowest is in January.
* From January to July it is upward trend and August to December is downward trend.

In [3]:
df.head()

,Month,Tourist_Arrivals,Country,Average_Travel_Cost,Hotel_Bookings,Hotel_Price,Season,Start_Date,End_Date,Activity,Hotel_Rating
0,Jan,12000,USA,800,300,100,Winter,2025-01-05,2025-01-10,Sightseeing,4.5
1,Feb,15000,UK,750,280,120,Winter,2025-02-10,2025-02-15,Adventure Sports,4.2
2,Mar,18000,Germany,900,320,130,Spring,2025-03-15,2025-03-20,Cultural Tour,4.7
3,Apr,21000,France,850,310,110,Spring,2025-04-20,2025-04-25,Beach Holiday,4.3
4,May,25000,India,700,290,140,Summer,2025-05-25,2025-05-30,Mountain Trek,4.1


# **Bar Plot**

* **Concept**
* A bar chart compares categories, making it suitable for analyzing top source countries for tourism.
* **Tourist Arrivals by Country**

In [6]:
fig = px.bar(df, x='Country', y='Tourist_Arrivals', title = "Tourist Arrivals by Country")
fig.show()


# **Scatter Plot**
* **Concept**
* A **scatter plot** helps visualize relationships between **two numerical variables**.


* **Hotel Bookings vs. Travel Cost**

In [11]:
ab = px.scatter(df, x='Average_Travel_Cost', y="Hotel_Bookings", title='Hotel Booking vs.Travel Cost')
ab.show()


## **Inference:**

* If bookings **increase with cost**, tourists may prefer luxury travel.
* If **inverse correlation**, budget-conscious tourism dominates.

# **Histogram**

**Concept:**

* A **histogram** shows the frequency distribution of numerical data, making it useful for analyzing hotel pricing.

**Distribution of Hotel Prices**

In [12]:
fig = px.histogram(df, x='Hotel_Price', title = 'Distribution of Hotel Prices')
fig.show()


## **Inference:**

* Identifies whether hotels are **affordable or expensive**.
* Peaks show the **most common price range**.

# Violin Plot

### Concept

* A **violin plot** combines a **box plot** and **KDE** (Kernel Density Estimation) to show **data distribution with density estimation**.

### Seasonal Variation in Travel Cost

---

This provides a helpful guide on how to build and interpret violin plots in Python to visualize data distributions effectively.

In [13]:
ab = px.violin(df, x='Season', y='Average_Travel_Cost', box= True, title='Seasonal Variation in Travel Cost')
ab.show()


## Inference

* Wider areas indicate **most frequent travel costs**.
* Outliers help **identify premium or budget travel trends**.
* **Highest Travel Cost**: Summer has the highest average travel cost at approximately 1000 units, making it the most expensive season for travel.
* **Lowest Travel Cost**: Winter has the lowest average travel cost at around 500 units, making it the most affordable season for travel.
* **Trend**: Travel costs rise steadily from Winter to Summer, peak in Summer, and then decline in Autumn, suggesting a cyclical pattern tied to seasonal demand (e.g., summer vacations increasing prices).


# **Gantt Chart**
* **Concept:** A **Gantt chart** visualizes schedules, making it ideal for **trip planning**. used in operational research.
* **Travel Itinerary Planning**

In [14]:
ba = px.timeline(df, x_start= 'Start_Date', x_end='End_Date', y='Activity', title='Tourism Itinary')
ba.show()


**Inference:**

* **Seasonal Activities:** The itinerary includes seasonal activities like "Skiing" (likely in winter months like Jan or Nov) and "Beach Holiday" (likely in summer months like Jul), suggesting travel plans vary by weather and time of year.
* **Diverse Options:** The list covers a wide range of interests (adventure, culture, relaxation), indicating a flexible or multi-purpose travel plan spanning 2025–2026.
* **Possible Gaps:** Months like Feb, Apr, Jun, etc., are missing, implying either no travel planned or focus on specific periods.

# Heatmap

### Concept

* A **heatmap** is useful for **finding patterns and correlations**.

In [15]:
import plotly.figure_factory as ff
import numpy as np

In [17]:
import plotly.graph_objects as go

# Compute correlation matrix (numerical columns only)
correlation_matrix = df.select_dtypes(include=['number']).corr()

# Create annotated heatmap using go.Heatmap
fig = go.Figure(data=go.Heatmap(
    z=correlation_matrix.values,
    x=correlation_matrix.columns.tolist(),
    y=correlation_matrix.index.tolist(),
    colorscale='Viridis',
    showscale=True,
    hoverongaps=False,
    text=[[f"{val:.2f}" for val in row] for row in correlation_matrix.values],
    texttemplate="%{text}",
))

# Update layout for spacing and readable text
fig.update_layout(
    title="Correlation Heatmap",
    width=800,
    height=800,
    margin=dict(t=50, l=100),
    font=dict(size=10)
)

fig.show()


## Inference

* **Inverse Relationship:** Higher `Hotel_Rating` (1.0) correlates with lower `Tourist_Arrivals` (0.098), suggesting premium-rated hotels may attract fewer, possibly high-end travelers.
* **Price vs. Demand:** The highest `Hotel_Price` (1.0) has mid-range `Hotel_Bookings` (0.461), indicating price sensitivity—luxury stays may not always maximize occupancy.
* **Travel Cost Impact:** Peak `Average_Travel_Cost` (1.0) aligns with high `Hotel_Bookings` (0.911), implying popular seasons (e.g., summer) drive both costs and demand.
* **Key Insight:** Balance between affordability (price/bookings) and exclusivity (rating/arrivals) is critical in tourism trends.

# 3D Plot

### Concept

* A **3D scatter plot** provides a **multi-dimensional view of relationships**.

### Travel Cost vs. Hotel Ratings & Bookings




In [18]:
fig = px.scatter_3d(df, x='Average_Travel_Cost', y='Hotel_Rating', z='Hotel_Bookings',
                    title='3D Analysis of Travel Cost, Ratings & Bookings', color='Hotel_Rating')
fig.show()


## Inference

* Helps compare **luxury vs. budget travel trends**.
* Identifies clusters of **popular hotel price ranges**.
* **Positive Correlation Between Hotel Rating and Bookings:** Higher-rated hotels (e.g., 4.8–4.9) tend to have more bookings, indicating that customers prioritize quality and ratings when choosing accommodations.
* **Travel Cost Impact on Bookings:** Mid-range travel costs may correlate with higher bookings, as customers likely balance affordability with quality. Extremely high or low travel costs could show fewer bookings, suggesting a preference for value-for-money options.
* **Rating Consistency Matters:** Hotels with ratings around 4.4–4.5 appear frequently, suggesting this range is common or optimal for attracting bookings while maintaining competitive pricing. Lower ratings (e.g., 4.0–4.2) may indicate fewer bookings or less customer preference.